internal/agent/client/podman.go

```diff
@@ -314,6 +314,113 @@ func (p *Podman) InspectArtifactAnnotations(ctx context.Context, reference strin
 	return extractArtifactAnnotations(inspect), nil
 }
 
+// ListImages returns a list of all container images stored on the device.
+// Returns image references in the format "repository:tag" or image ID for untagged images.
+func (p *Podman) ListImages(ctx context.Context) ([]string, error) {
+	ctx, cancel := context.WithTimeout(ctx, p.timeout)
+	defer cancel()
+
+	// Use a format that handles both tagged and untagged images
+	// For tagged images: "repository:tag"
+	// For untagged images: use the image ID (short format)
+	// Note: When Repository is the string "<none>" (not empty), we need to check for it explicitly
+	args := []string{"image", "ls", "--format", "{{if and .Repository (ne .Repository \"<none>\")}}{{.Repository}}:{{.Tag}}{{else}}{{.ID}}{{end}}"}
+	stdout, stderr, exitCode := p.exec.ExecuteWithContext(ctx, podmanCmd, args...)
+	if exitCode != 0 {
+		return nil, fmt.Errorf("list images: %w", errors.FromStderr(stderr, exitCode))
+	}
+
+	lines := strings.Split(strings.TrimSpace(stdout), "\n")
+	imagesSeen := make(map[string]struct{})
+	images := make([]string, 0, len(lines))
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if line == "" {
+			continue
+		}
+		if _, ok := imagesSeen[line]; !ok {
+			imagesSeen[line] = struct{}{}
+			images = append(images, line)
+		}
+	}
+
+	return images, nil
+}
+
+// ListArtifacts returns a list of all OCI artifacts stored on the device.
+func (p *Podman) ListArtifacts(ctx context.Context) ([]string, error) {
+	ctx, cancel := context.WithTimeout(ctx, p.timeout)
+	defer cancel()
+
+	if err := p.EnsureArtifactSupport(ctx); err != nil {
+		return nil, err
+	}
+
+	args := []string{"artifact", "ls", "--format", "{{.Name}}"}
```

Comment:
we need to be careful with artifact support what version of podman was this added? can we review that with what is shipped with RHEL 9.5+. we don't want issues where support is not available in some versions uncontrolled. this probably is true for other areas of the client but lets verify :pray: